rdr_service/resource/generators/participant.py

```diff
@@ -1662,7 +1666,7 @@ def _find_consent_response(self, response_rec: BQRecord, module: str):
                   consent_status:  BQModuleStatusEnum
         """
         # If a module doesn't have defined consent questions that determine status, it defaults to SUBMITTED status
-        if _consent_module_question_map[module] is None:
+        if module in _consent_module_question_map and _consent_module_question_map[module] is None:
```

Comment:
@kenny-skaggs This line was intended to be specific to modules that must be a key in the map (must be a recognized consent module), but have a None value because there is no explicit consent question (e.g., ConsentPII, ProgramUpdate).   On line 740, the initialization was intending to default any non-consent modules (e.g., PPI modules), or consent modules without a question code, to SUBMITTED.

But I'll revisit this to see if I can make this less confusing.   This`find_consent_response()`  function  did not exist originally, but was added in a prior refactor and appears to have overlapped with the logic on line 740.

Reply:
My mistake!

Reply:
@kenny-skaggs You pointed out some redundant logic that should have been  handled better in previous refactorings.  We're avoiding any major updates to this code and deferring to the new pipeline development, but I've made some minor changes that I hope will make the logic a little more clear.  Thanks!